authz.go

```diff
@@ -184,10 +184,18 @@ func (a *authorizer) authorize(ctx context.Context, sr authz.SubjectAccessReview
 	var via string
 	var client *client
 	if a.AuthorizationConfig.Config.UseCache {
-		decision := a.useCacheEval(log, principal, checks)
-		if decision != nil && !a.AuthorizationConfig.Config.DryRun {
-			return decision
+		// check syncer's last contact time with athenz, if it is more than two hours,
+		// fall back to zms/zts.
+		a.AuthorizationConfig.Config.Cache.lock.Lock()
```

Comment:
Can we make have this be a read lock here? We can store the contents of cacheStatus into a temporary variable and unlock right away instead of unlocking after the entire if logic.

Reply:
Let's also make this a separate lock.